.github/workflows/test-upstream.yml

```diff
@@ -1,14 +1,28 @@
 name: Nightly upstream testing
 on:
   schedule:
-    - cron: "0 0 * * *" # Daily “At 00:00” UTC
+    - cron: "0 0 * * *" # Daily “At 00:00” UTC for upstream dask testing
+    - cron: "0 3 * * *" # Daily "At 03:00" UTC for upstream datafusion testing
   workflow_dispatch: # allows you to trigger the workflow run manually
-
+    inputs:
+      upstreamLib:
+        type: choice
+        description: 'Library to update for upstream testing'
+        required: false
+        default: 'Dask'
+        options:
+          - Dask
+          - DataFusion
 # Required shell entrypoint to have properly activated conda environments
 defaults:
   run:
     shell: bash -l {0}
 
+env:
+  which_upstream: ${{ (github.event.schedule == '0 3 * * *' && 'DataFusion') ||
+                    (github.event.schedule == '0 0 * * *' && 'Dask') ||
+                    (github.event.inputs.upstreamLib)}}
```

Comment:
This works fine but just as an FYI think we can format multi-line expressions like so:

```suggestion
  which_upstream: |
    (github.event.schedule == '0 3 * * *' && 'DataFusion')
    || (github.event.schedule == '0 0 * * *' && 'Dask')
    || (github.event.inputs.upstreamLib)
```